src/common/snippets/src/pass/softmax_decomposition.cpp

```diff
@@ -0,0 +1,231 @@
+    register_matcher(std::make_shared<ngraph::pattern::Matcher>(
+        ngraph::pattern::wrap_type<ngraph::op::v1::Softmax, ngraph::op::v8::Softmax>(), matcher_name),
+        [=](ngraph::pattern::Matcher &m) {
+        OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::SoftmaxDecomposition")
+        auto root = m.get_match_root();
+        const auto master_pshape = root->get_input_partial_shape(0);
+        const auto rank = master_pshape.rank();
+        if (rank.is_dynamic() || master_pshape.is_dynamic())
+            return false;
+
+        int64_t axis = 0;
+        if (const auto softmax_v8 = ngraph::as_type_ptr<const ov::op::v8::Softmax>(root)) {
+            axis = ngraph::normalize_axis(root->get_friendly_name(), softmax_v8->get_axis(), rank);
+        } else if (const auto softmax_v1 = ngraph::as_type_ptr<const ov::op::v1::Softmax>(root)) {
+            axis = softmax_v1->get_axis();
+        } else {
+            return false;
+        }
+
+        const auto shape_rank = rank.get_length();
+        if (axis != shape_rank - 1)
+            return false;
+
+        const auto data = root->get_input_node_shared_ptr(0);
+        const bool input_is_buffer = ov::is_type<ngraph::snippets::op::Buffer>(data);
+        // After Softmax can be inserted only one common Buffer if needed
+        const bool output_is_buffer = ov::is_type<ngraph::snippets::op::Buffer>(root->output(0).get_target_inputs().begin()->get_node());
+
+        const auto master_shape = master_pshape.get_shape();
+        const auto dimension = shape_rank - 1;
+        const auto work_amount = master_shape[dimension];
+        const auto increment = vector_size;
+        const auto inner_dim = shape_rank - 1;
+        const auto inner_size = static_cast<size_t>(master_shape[inner_dim]);
+        const int outer_dim = shape_rank > 1 ? shape_rank - 2 : -1;
+        const auto has_outer_loop = outer_dim >= 0 && master_shape[outer_dim] > 1;
+
+        /* ====== ReduceMax decomposition ====== */
+
+        const auto vector_buffer_max = std::make_shared<ngraph::snippets::op::VectorBuffer>();
+        const auto loop_max_begin = ngraph::snippets::op::insertLoopBegin(ngraph::OutputVector{data, data});
+
+        const auto load_max = std::make_shared<ngraph::snippets::op::Load>(loop_max_begin->output(0), increment);
+        const auto max = std::make_shared<ov::op::v1::Maximum>(load_max, vector_buffer_max);
+
+        const auto apply_increments_max = std::vector<bool>{ calculate_apply_increment(inner_size, data->get_shape()[inner_dim]), false, false };
+        // we should always reset data ptr after this loop because in the next Loop this ptr is used
+        const auto finalization_offsets_max =
+            std::vector<int64_t>{ calculate_required_finalization_offsets(inner_size, data->get_shape()[inner_dim]), 0, 0 };
+        const auto forse_finalization_offsets_max = std::vector<bool>{true, false, false};
```

Comment:
To continue `force_finalization_offsets` discussion: here is a good example, why can't we just set two last finalization_offsets to zero instead of setting force_finalization_offsets to false?

Reply:
As we discussed offline - I disabled `one evaluation optimizations` for subgraphs with buffers to remove `force_finalization`